docs/interface.md

```diff
@@ -66,28 +66,8 @@ modes, but not learn the identity of the chosen application):
 * The data could not be delivered to the target app (e.g., the chosen app could
   not be launched), or the target app explicitly rejected the share event.
 
-## navigator.canShare
-
-`navigator` also provides a method for determining whether there are any
-applications that can handle sharing:
-
-```WebIDL
-partial interface Navigator {
-  boolean canShare();
-};
-```
-
-Returns `true` if there are one or more applications that could handle a share
-event (i.e., if `share` was called, would any applications be presented to the
-user?). May give false positives, but not false negatives (on some systems, it
-may not be possible to determine in advance whether any native applications
-support sharing, in which case `canShare` should return `true`; `false` means
-that `share` will definitely fail). This can be used by websites to hide or
-disable the sharing UI, to avoid presenting a button that just fails when users
-press it.
-
-**TODO(mgiuca)**: This may have to be asynchronous, so that the implementation
-can query the file system without blocking.
+The `navigator.share` method should be `undefined` if the user agent does not
```

Comment:
The proposal here is not very idiomatic and would cause it to behave unlike any other DOM API. Share targets can be added asynchronously at any point in time (via the OS or via a share target being registered). As such, this should probably not appear and disappear from one turn of the event loop to another (or change from a function to undefined. The problem is that `navigator.hasOwnProperty("share")` would return `true` even if set to `undefined` ... which would be confusing). 

We could work on the assumption that there is always at least 1 share target: the UA share box itself... For instance, Safari has "Add to Reading List" and "Add To Bookmarks" as persistent share targets. Other UAs might inform the user about adding share targets or might simply allow the user to cancel.